File: app/src/main/java/com/tunjid/androidbootstrap/viewmodels/RouteViewModel.java
package com.tunjid.androidbootstrap.viewmodels;

import android.app.Application;
import android.content.Context;

import com.tunjid.androidbootstrap.R;
import com.tunjid.androidbootstrap.core.text.SpanBuilder;
import com.tunjid.androidbootstrap.fragments.BleScanFragment;
import com.tunjid.androidbootstrap.fragments.DoggoListFragment;
import com.tunjid.androidbootstrap.fragments.DoggoRankFragment;
import com.tunjid.androidbootstrap.fragments.HidingViewFragment;
import com.tunjid.androidbootstrap.fragments.NsdScanFragment;
import com.tunjid.androidbootstrap.fragments.SpanbuilderFragment;
import com.tunjid.androidbootstrap.fragments.TileFragment;
import com.tunjid.androidbootstrap.model.Route;

import java.util.Arrays;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.StringRes;
import androidx.lifecycle.AndroidViewModel;

public class RouteViewModel extends AndroidViewModel {

    private final List<Route> routes;

    public RouteViewModel(@NonNull Application application) {
        super(application);
        routes = Arrays.asList(
                new Route(DoggoListFragment.class.getSimpleName(), formatRoute(R.string.route_doggo_list)),
                new Route(DoggoRankFragment.class.getSimpleName(), formatRoute(R.string.route_doggo_rank)),
                new Route(TileFragment.class.getSimpleName(), formatRoute(R.string.route_tile)),
                new Route(HidingViewFragment.class.getSimpleName(), formatRoute(R.string.route_hiding_view)),
                new Route(SpanbuilderFragment.class.getSimpleName(), formatRoute(R.string.route_span_builder)),
                new Route(BleScanFragment.class.getSimpleName(), formatRoute(R.string.route_ble_scan)),
                new Route(NsdScanFragment.class.getSimpleName(), formatRoute(R.string.route_nsd_scan))
        );
    }

    public List<Route> getRoutes() {
        return routes;
    }

    private CharSequence formatRoute(@StringRes int stringRes) {
        Context context = getApplication();
        return SpanBuilder.of(context.getString(stringRes))
                .italic()
                .underline()
                .color(context, R.color.colorPrimary)
                .build();
    }
}

File: app/src/main/java/com/tunjid/androidbootstrap/viewmodels/RouteViewModel.kt
package com.tunjid.androidbootstrap.viewmodels

import android.app.Application
import androidx.annotation.IdRes
import androidx.annotation.StringRes
import androidx.lifecycle.AndroidViewModel
import com.tunjid.androidbootstrap.R
import com.tunjid.androidbootstrap.core.text.SpanBuilder
import com.tunjid.androidbootstrap.fragments.*
import com.tunjid.androidbootstrap.model.Route

class RouteViewModel(application: Application) : AndroidViewModel(application) {

    private val mapping = mutableMapOf(
            R.id.menu_core to listOf(
                    Route(DoggoListFragment::class.java.simpleName, formatRoute(R.string.route_doggo_list)),
                    Route(HidingViewFragment::class.java.simpleName, formatRoute(R.string.route_hiding_view)),
                    Route(SpanbuilderFragment::class.java.simpleName, formatRoute(R.string.route_span_builder))
            ),
            R.id.menu_recyclerview to listOf(
                    Route(DoggoRankFragment::class.java.simpleName, formatRoute(R.string.route_doggo_rank)),
                    Route(ShiftingTileFragment::class.java.simpleName, formatRoute(R.string.route_shifting_tile)),
                    Route(EndlessTileFragment::class.java.simpleName, formatRoute(R.string.route_endless_tile))
            ),
            R.id.menu_communications to listOf(
                    Route(BleScanFragment::class.java.simpleName, formatRoute(R.string.route_ble_scan)),
                    Route(NsdScanFragment::class.java.simpleName, formatRoute(R.string.route_nsd_scan))
            )
    )

    fun getRoutes(@IdRes id: Int): List<Route> = mapping[id] ?: mapping.values.flatten()

    private fun formatRoute(@StringRes stringRes: Int): CharSequence = getApplication<Application>().run {
        SpanBuilder.of(getString(stringRes))
                .italic()
                .underline()
                .color(this, R.color.colorPrimary)
                .build()
    }
}
